src/clib/pio_rearrange.c

```diff
@@ -1229,6 +1229,276 @@ int box_rearrange_create(iosystem_desc_t *ios, int maplen, const PIO_Offset *com
     MPI_Datatype dtypes[ios->num_uniontasks]; /* Array of MPI_OFFSET types for swapm. */
     PIO_Offset iomaplen[ios->num_iotasks];   /* Gets the llen of all IO tasks. */
 
+    /* sc_info msg = [iomaplen, starts_for_all_dims, count_for_all_dims] */
+    int sc_info_msg_maplen_sz = 1; /* The iomaplen, == 0 implies start/count are invalid */
+    int sc_info_msg_sc_sz = 2 * ndims; /* The (start + count) for all dims */
+    int sc_info_msg_sz = sc_info_msg_maplen_sz + sc_info_msg_sc_sz;
+    PIO_Offset sc_info_msg_send[sc_info_msg_sz];
+    PIO_Offset sc_info_msg_recv[ios->num_iotasks * sc_info_msg_sz];
+
+    /* This is the box rearranger. */
+    iodesc->rearranger = PIO_REARR_BOX;
+
+    /* Number of elements of data on compute node. */
+    iodesc->ndof = maplen;
+
+    if (maplen > 0)
+    {
+        if (!(dest_ioproc = malloc(maplen * sizeof(int))))
+            return pio_err(ios, NULL, PIO_ENOMEM, __FILE__, __LINE__);
+
+        if (!(dest_ioindex = malloc(maplen * sizeof(PIO_Offset))))
+            return pio_err(ios, NULL, PIO_ENOMEM, __FILE__, __LINE__);
+    }
+
+    /* Initialize the sc_info send and recv messages */
+    for(int i=0; i<sc_info_msg_sz; i++)
+    {
+        sc_info_msg_send[i] = 0;
+    }
+
+    /* The sc_info_msg_recv[i * sc_msg_info_sz] contains the sc_info from
+     * iorank i (the union rank for iorank i is ios->ioranks[i]). Each
+     * sc_info message contains [iomaplen, start_for_all_dims, count_for_all_dims]
+     */
+    for(int i=0; i<ios->num_iotasks * sc_info_msg_sz; i++)
```

Comment:
Same as above, memset can be used here.

Reply:
We could perform such optimizations in another PR, if needed (most likely the compiler will do it for us).